pkg/cache/sql/store/store.go

```diff
@@ -17,17 +17,16 @@ import (
 )
 
 const (
-	upsertStmtFmt   = `REPLACE INTO "%s"(key, object, objectnonce, dek, deknonce) VALUES (?, ?, ?, ?, ?)`
+	upsertStmtFmt   = `REPLACE INTO "%s"(key, object, objectnonce, dekid) VALUES (?, ?, ?, ?)`
 	deleteStmtFmt   = `DELETE FROM "%s" WHERE key = ?`
-	getStmtFmt      = `SELECT object, objectnonce, dek, deknonce FROM "%s" WHERE key = ?`
-	listStmtFmt     = `SELECT object, objectnonce, dek, deknonce FROM "%s"`
+	getStmtFmt      = `SELECT object, objectnonce, dekid FROM "%s" WHERE key = ?`
+	listStmtFmt     = `SELECT object, objectnonce, dekid FROM "%s"`
 	listKeysStmtFmt = `SELECT key FROM "%s"`
 	createTableFmt  = `CREATE TABLE IF NOT EXISTS "%s" (
 		key TEXT UNIQUE NOT NULL PRIMARY KEY,
 		object BLOB,
-		objectnonce TEXT,
-		dek TEXT,
-		deknonce TEXT
+		objectnonce BLOB,
+		dekid BLOB
```

Comment:
What pros and cons do you see in using a UUID for the DEK id rather than an incremental counter?

How about storing DEKs in a slice, appending on rotation, and using the slice index as the key identifier instead of the UUID?

Reply:
Funnily enough, that was my initial approach: the numerical counter part. I pivoted from that mostly to get a step closer to parallel key creation without needing to change the DB schema. I will revert back to that approach instead.